artifactory/services/utils/resultutils.go

```diff
@@ -53,13 +53,20 @@ func (ad *ArtifactDetails) ToBuildInfoArtifact() buildinfo.Artifact {
 	artifact := buildinfo.Artifact{Checksum: buildinfo.Checksum{}}
 	artifact.Sha1 = ad.Checksums.Sha1
 	artifact.Md5 = ad.Checksums.Md5
+	artifact.Sha256 = ad.Checksums.Sha256
 	// Artifact name in build info as the name in artifactory
 	filename, _ := fileutils.GetFileAndDirFromPath(ad.ArtifactoryPath)
 	artifact.Name = filename
 	if i := strings.LastIndex(filename, "."); i != -1 {
 		artifact.Type = filename[i+1:]
 	}
-	artifact.Path = ad.ArtifactoryPath
+	// Remove the repository from Artifact's path.
+	// Since a build can be promoted, the build-info may be incorrect (builds are immutable).
```

Comment:
I think it's best to simplify the above to - 

// The 'path' property in the build-info should not include the repository. We therefore remove the repository from the path.